builtin/logical/ssh/path_sign.go

```diff
@@ -473,31 +473,36 @@ func (b *backend) validateSignedKeyRequirements(publickey ssh.PublicKey, role *s
 			return fmt.Errorf("pubkey not suitable for crypto (expected ssh.CryptoPublicKey but found %T)", k)
 		}
 
-		if value, ok := role.AllowedUserKeyLengths[kstr]; ok {
+		if allowed_values, ok := role.AllowedUserKeyTypesLengths[kstr]; ok {
 			var pass bool
-			switch kstr {
-			case "rsa":
-				if kbits == value {
-					pass = true
-				}
-			case "dsa":
-				if kbits == value {
-					pass = true
-				}
-			case "ecdsa":
-				if kbits == value {
+			for _, value := range allowed_values {
+				switch kstr {
+				case "rsa":
```

Comment:
Technically correct, but we are now incongruent on the key names we accept for "key types" from PR#14008 and what the names/values we allow for this key. Not the end of the world just pointing it out.